nservicebus/upgrades/azure-functions-service-bus-in-process-isolated-worker.md

```diff
@@ -146,3 +146,27 @@ class HttpSender
     }
 }
 ```
+
+## Ensuring consistency in the Isolated Worker model
+
+It's important to be able to maintain the consistency guarantees that were previously available in the in-process model when using `SendsAtomicWithReceive`.  To achieve this, there are a few options:
+
+1. **Implement the [Outbox pattern](/architecture/consistency.md#transactions-outbox-pattern)** to simulate atomic transactions. (Recommended)
+2. Ensure that all receiver message handlers are [idempotent](/architecture/consistency.md#idempotency).
+3. Ensure that the system infrastructure guarantees consistency between business data and messages.
```

Comment:
I'm not sure about this one, do U have some examples of this when running isolated workers?

Reply:
Maybe this is my misunderstanding, but we are recommending using Outbox and in the documation for [Outbox](https://docs.particular.net/nservicebus/outbox/) it states that the outbox feature ensures consistency between business data and messages.  Even in [the consistency problem](https://docs.particular.net/nservicebus/outbox/#the-consistency-problem) it states that there are two ways to avoid zombie and ghost messages and one of the ways is to implement infrastructure which guarantees consistency, which describes the outbox feature.  I'm assuming that if the system infrastructure can guarantee consistency between business data and messages, then the outbox feature wouldn't be unnecessary?  Obviously, to me, this seems like the most difficult solution to implement from a users perspective, especially if all they had to do before in the in-process model is add the `SendsAtomicWithReceive` assembly attribute.

Reply:
Yes, if the infrastructure can handle this for you (eg, using MSDTC or our SQL transport together with storing business data in SQL server), this is a good option as mentioned. 

That said in this case, using functions over Azure ServiceBus there is no infrastructure solution available (and I can't see one ever being provided by MS) so in this context I don't think it's an option for the users.

Does this make sense?